src/posix/client.cpp

```diff
@@ -181,23 +181,61 @@ int main(int argc, char *argv[])
     {
         char   buffer[kLineBufferSize];
         size_t count = 0;
+        char **arg   = &argv[1];
 
-        for (int i = 1; i < argc; i++)
+        enum
         {
-            for (const char *c = argv[i]; *c; ++c)
+            kStateArgument,
+            kStateEscaping,
+            kStateEnd,
+        } state = kStateArgument;
+
+        --argc;
+
+        switch (state)
+        {
+            do
             {
-                if (IsSeparator(*c))
+            case kStateEnd:
```

Comment:
+1 suggest against `switch { do { case ...: ... } while(...); }` pattern.

@zhanglongxia

Reply:
I think we may need to improve the code read readability here. This code pattern is not friendly to the code readers.

Reply:
Readability matters.

Despite a small program, src/posix/client.cpp (i.e. ot-ctl) has a history of subtle bugs:
- #5087
- #5134
- #6378
- This PR

I think one reason is that we are trying too hard to make it as efficient as possible, meanwhile it actually does not matter at all for a command like `ot-ctl`.